packages/node/src/core/evm/abi-encoding.test.ts

```diff
@@ -0,0 +1,28 @@
+import { safeDecode } from './abi-encoding';
+
+describe('safeDecode', () => {
+  it('decodes parameters successfully', () => {
+    const res = safeDecode(
+      '0x315375000000000000000000000000000000000000000000000000000000000066726f6d0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0616d6f756e74000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000003e800000000000000000000000000000000000000000000000000000000000000034554480000000000000000000000000000000000000000000000000000000000'
+    );
+    expect(res).toEqual({
+      from: 'ETH',
+      amount: '1000',
+    });
+  });
+
+  it('returns null if parameters are empty', () => {
+    const res = safeDecode('0x');
+    expect(res).toEqual(null);
+  });
+
+  it('returns an empty object if parameters are falsey', () => {
+    const res = safeDecode('');
+    expect(res).toEqual({});
```

Comment:
and `null` here because `"0x"` is a valid empty encoding and `""` is not

Reply:
For context, if you declare a `bytes` variable in a contract and retrieve it before initializing it, it returns `"0x"`